samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -348,7 +349,9 @@ private JobModel generateNewJobModel(List<String> processors) {
      * Host affinity is not supported in standalone. Hence, LocalityManager(which is responsible for container
      * to host mapping) is passed in as null when building the jobModel.
      */
-    return JobModelManager.readJobModel(this.config, changeLogPartitionMap, null, streamMetadataCache, processors);
+    JobModel model = JobModelManager.readJobModel(this.config, changeLogPartitionMap, null, streamMetadataCache, processors);
+    // Nuke the configuration in JobModel.
+    return new JobModel(new MapConfig(), model.getContainers());
```

Comment:
Assert it in an existing unit-test maybe?

Reply:
Done.